cpp/src/arrow/compute/exec_internal.h

```diff
@@ -29,6 +29,7 @@
 #include "arrow/compute/kernel.h"
 #include "arrow/status.h"
 #include "arrow/util/visibility.h"
+#include "arrow/visit_data_inline.h"
```

Comment:
Is there a more specific header for `::arrow::internal::call_traits::enable_if_return`?